crates/admin/src/cluster_controller/service.rs

```diff
@@ -61,22 +58,21 @@ pub enum Error {
 }
 
 pub struct Service<T> {
-    task_center: TaskCenter,
-    health_status: HealthStatus<AdminStatus>,
-    metadata: Metadata,
-    networking: Networking<T>,
-    bifrost: Bifrost,
-    cluster_state_refresher: ClusterStateRefresher<T>,
+    pub(crate) task_center: TaskCenter,
+    pub(crate) metadata: Metadata,
+    pub(crate) networking: Networking<T>,
+    pub(crate) bifrost: Bifrost,
+    pub(crate) cluster_state_refresher: ClusterStateRefresher<T>,
+    pub(crate) configuration: Live<Configuration>,
+    pub(crate) metadata_writer: MetadataWriter,
+    pub(crate) metadata_store_client: MetadataStoreClient,
```

Comment:
If `cluster_controller_state` were a module of `service`, then I believe you don't have to make these fields `pub(crate)`.